### PR TITLE
converted cube directory tests to pytest

### LINKED ISSUE #5821: Convert unit/cube test files to pytest

### PR DESCRIPTION
**Pull request by ESadek-MO**
closes #5821

### CODE REVIEW

File: lib/iris/tests/unit/cube/test_Cube.py [hunk 1]
```diff
@@ -1048,30 +1062,30 @@ def test_1d_slice_coord_given(self):
...
     def test_1d_slice_dimension_given(self):
         res = self.cube.slices_over(1)
         for i, res_cube in zip(self.exp_iter_1d, res):
             expected = self.cube[:, i]
-            self.assertEqual(res_cube, expected)
+            assert res_cube == expected
 
     def test_1d_slice_nonexistent_dimension_given(self):
-        with self.assertRaisesRegex(ValueError, "iterator over a dimension"):
+        with pytest.raisesRegex(ValueError, "iterator over a dimension"):
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_Cube.py:1088]: It's still `pytest.raises()`. I _believe_ the `match` argument will work with regex if provided, and also with a partial text match as seen here. Just check your arguments basically, but it's probably fine!

**Review by trexfeathers — CHANGES_REQUESTED** [review]
I've checked your working all the way through and these are comments are all I have ⭐

**Comment by trexfeathers** [comment]
Please remember to run `ruff --select=PT lib/iris/tests/unit/cube/test_Cube.py` before you are done

File: lib/iris/tests/unit/cube/test_Cube.py (hunk 1 above)
Comment (ESadek-MO) [reply lib/iris/tests/unit/cube/test_Cube.py:1088]: Yes, you're right! Just missed this one, thanks for the catch!

File: lib/iris/tests/unit/cube/test_CubeAttrsDict.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 from iris.fileformats.netcdf.saver import _CF_GLOBAL_ATTRS
 
 
-@pytest.fixture
+@pytest.fixture()
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_CubeAttrsDict.py:17]: The brackets are unnecessary ([based on the docs](https://docs.pytest.org/en/stable/explanation/fixtures.html)), but I don't know if they can actually cause problems.

File: lib/iris/tests/unit/cube/test_Cube__operators.py [hunk 3]
```diff
@@ -144,7 +145,7 @@ def build_lazy_cube(self, value, dtype=np.float64):
         data = as_lazy_data(np.array(value, dtype=dtype))
         return iris.cube.Cube(data, standard_name="air_temperature", units="K")
 
-    def setUp(self):
+    def setup_method(self):
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_Cube__operators.py]: Missed a spot

File: lib/iris/tests/unit/cube/test_Cube.py [hunk 4]
```diff
@@ -489,8 +508,9 @@ def test_weighted_sum_with_unknown_units_lazy_y(self):
 class Test_collapsed__multidim_weighted_with_cube(
     Test_collapsed__multidim_weighted_with_arr
 ):
-    def setUp(self):
-        super().setUp()
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        super()._setup()
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_Cube.py:513]: You're not supposed to call fixtures directly. If you need to inherit some behaviour then I would give this fixture a new name like `_setup_subclass` and then pass `_setup` as an argument (I have some examples in my PRs)

**Review by trexfeathers — CHANGES_REQUESTED** [review]

File: lib/iris/tests/unit/cube/test_Cube.py (hunk 4 above)
Comment (trexfeathers) [reply lib/iris/tests/unit/cube/test_Cube.py:513]: Had this idea on Friday:

```python
@pytest.fixture(autouse=True)
def _setup(self, ParentClass._setup):
   ...
```

File: lib/iris/tests/unit/cube/test_Cube__operators.py [hunk 5]
```diff
@@ -7,16 +7,17 @@
...
 import numpy.ma as ma
+import pytest
 
 import iris
 from iris._lazy_data import as_lazy_data
 from iris.coords import DimCoord
+from iris.tests import _shared_utils
 
 
 class Test_lazy_maths(tests.IrisTest):
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_Cube__operators.py:23]: Still inheriting from `IrisTest` here

File: lib/iris/tests/unit/cube/test_CubeList.py [hunk 6]
```diff
@@ -21,31 +23,34 @@
...
 import iris.exceptions
 from iris.fileformats.pp import STASH
+from iris.tests import _shared_utils
 import iris.tests.stock
 
 NOT_CUBE_MSG = "cannot be put in a cubelist, as it is not a Cube."
 NON_ITERABLE_MSG = "object is not iterable"
 
 
 class Test_append(tests.IrisTest):
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_CubeList.py:33]: Need to stop inheriting from `IrisTest`

File: lib/iris/tests/unit/cube/test_Cube.py [hunk 7]
```diff
@@ -304,50 +306,51 @@ def setUp(self):
...
-        self.assertArrayEqual(result.data, np.mean(self.data, axis=1))
+        assert not result.has_lazy_data()
+        _shared_utils.assert_array_equal(result.data, np.mean(self.data, axis=1))
 
 
-class Test_collapsed__multidim_weighted_with_arr(tests.IrisTest):
-    def setUp(self):
+class Test_collapsed__multidim_weighted_with_arr:
+    @pytest.fixture(autouse=True)
+    def multidim_arr_setup(self):
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_Cube.py:353]:
```
PT004 Fixture `multidim_arr_setup` does not return anything, add leading underscore
```

File: lib/iris/tests/unit/cube/test_Cube.py [hunk 8]
```diff
@@ -1026,16 +1046,17 @@ def check_order(self, dim1, dim2, dim3, dim_to_remove):
...
         expected_cube = self.expected_cube_setup(dim1, dim2, dim3)
-        self.assertEqual(sliced_cube, expected_cube)
+        assert sliced_cube == expected_cube
 
     def test_all_permutations(self):
         for perm in permutations(["time", "height", "latitude", "longitude"]):
             self.check_order(*perm)
 
 
 @tests.skip_data
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_Cube.py:1056]:
```suggestion
@_shared_utils.skip_data
```

File: lib/iris/tests/unit/cube/test_Cube__aggregated_by.py [hunk 9]
```diff
@@ -20,11 +21,13 @@
 import iris.coords
 from iris.coords import AncillaryVariable, AuxCoord, CellMeasure, DimCoord
 from iris.cube import Cube
+from iris.tests import _shared_utils
 from iris.tests.stock import realistic_4d
 
 
 class Test_aggregated_by(tests.IrisTest):
```
Comment (trexfeathers) [inline lib/iris/tests/unit/cube/test_Cube__aggregated_by.py:28]: Need to stop inheriting from `IrisTest`

**Review by trexfeathers — CHANGES_REQUESTED** [review]
I agree with every change you've made.

Just gave the files themselves the once-over and came up with these extra comments.

**Review by trexfeathers — APPROVED** [review]

**Pull request merged**